doc/0003-wizarding-api.md

```diff
@@ -0,0 +1,227 @@
+
+
+### API Specification
+
+In this section we define the lifecycle of a wizard plugin and describe each step.
+
+#### Structure
+
+```txt
+┌──────────────────────────────────┐
+│Core                              │
+│   ┌─────────────┐┌─────────────┐ │
+│   │Editor Plugin││Wizard Plugin│ │
+│   └─────────────┘└─────────────┘ │
+└──────────────────────────────────┘
+```
+
+We separate editors and wizards so that wizards can be used by other editors. 
+This, of course, could create a coupling between plugins, so a default or 
+fallback wizard is necessary to handle all elements e.g.: a generic XML editor
+
+#### Lifecycle
+
+```txt
+┌─────────────────┐ ┌───────────┐┌─────────────────┐     ┃
+│  Editor Plugin  │ │   Core    ││  Wizard Plugin  │     ┃  LEGEND
+└─────────────────┘ └───────────┘└─────────────────┘     ┃  ┌───────────┐
+         │                │               │              ┃  │  <name>   │   module with lifeline
+         │                │               │              ┃  └─────┬─────┘
+         │                │◀───register───│              ┃        │
+         │───register────▶│               │              ┃        │
+         │                │               │              ┃        │
+         │   requests a   │               │              ┃        │
+         │───wizard for──▶│               │              ┃
+         │    a given     │  asks wizard  │              ┃
+         │                │───if it can ─▶│              ┃  ──<desc.>  ─▶    Initiated action
+         │                │ handle given  │              ┃
+         │                │               │              ┃
+         │                │◀ ─ ─answer ─ ─│              ┃  ─ <answer>  ▶    Answer to an action
+         │                │               │              ┃
+         │                │   initiates   │              ┃
+         │                │    wizard     │              ┃            ┌─┐
+         │                │───with the ──▶│              ┃            │ │
+         │                │     given    ┌┴┐             ┃            │ │
+         │                │    element   │ │             ┃            │ │    Internal process
+         │                │              │ │wizarding    ┃            │ │
+         │                │              │ │process      ┃            │ │
+         │                │              │ │             ┃            └─┘
+         │                │              │ │             ┃
+         │                │              └┬┘             ┃
+         │                │               ├──┐           ┃
+         │                │               │  │           ┃
+         │                │               closes         ┃
+         │                │               itself         ┃
+         │                │               │  │           ┃
+         │                │               ◀──┘           ┃
+         │                │               │              ┃
+         X                X               X              ┃
```

Comment:
For invoking the next wizard on the deque (in order to know when to render a wizard plugin at all, and to move on to the next wizard) and for deciding when to stop rendering the wizard selector button (which will be needed to achieve feature parity with the current legacy open-scd implementation) we will need to know when the wizard is done with its work. An event that says "I'm done" would likely be needed here.

Reply:
would you prefer an event after the closure or an event that asks the core to close the wizard?

Reply:
I don't have a preference, but it's a decision I would really want us to think about the consequences of. Whether any wizards ever could need any cleanup, or whether "well, you're not being rendered any more now!" is always going to be acceptable as a way for core to close wizards. In the latter case I think I'd prefer that since it's simpler and doesn't require for the wizard itself to do any opening or closing, just to always render its stuff. What do you think?

Reply:
I agree and I have already made changes